tools/iree-dump-parameters-main.c

```diff
@@ -120,6 +120,12 @@ int main(int argc, char** argv) {
                        "Dumps information about parsed parameter files.\n");
   iree_flags_parse_checked(IREE_FLAGS_PARSE_MODE_DEFAULT, &argc, &argv);
 
+  if (argc > 1) {
+      fprintf(stderr, "Error, no positional arguments expected. \n"
+                      "Pass --parameters=flags\n");
+      exit_code = EXIT_FAILURE;
```

Comment:
you want to actually exit here instead of continuing to run and it's useful to print the flag in the form it's needed.
```suggestion
      fprintf(stderr, "Error: no positional arguments expected.\n");
      fprintf(stderr, "Use one or more --parameters=file.ext flags to specify parameter files.\n");
      IREE_TRACE_ZONE_END(z0);
      return EXIT_FAILURE;
```